### PR TITLE
Contributors with tags, randomized

### PR DESCRIPTION
**Pull request by hero101**
⚠️ This PR has to be merged after https://github.com/alkem-io/server/pull/4590

## Summary by CodeRabbit

- **New Features**
	- Introduced a new function to shuffle arrays, enhancing data handling.
	- Added a parameter to filter users based on tags in the contributors search.
	- Increased the number of contributors displayed per page from 16 to 32.
	- Added a conditional rendering for the search input field based on user interaction.

- **Bug Fixes**
	- Corrected a typo in the organizations query result.

- **Improvements**
	- Optimized user rendering by shuffling only new users added to the list.

### CODE REVIEW

File: src/core/utils/array.shuffle.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const arrayShuffle = <T>(array: T[]): T[] => {
+  const shuffledArray = array.slice(); // Create a copy of the array
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/core/utils/array.shuffle.ts:2]: the function is creating a new array without mutating the underlying objects

File: src/core/utils/array.shuffle.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  for (let i = shuffledArray.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/core/utils/array.shuffle.ts:4]: The function is just shuffling elements, nothing mission critical.

File: src/core/utils/array.shuffle.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    [shuffledArray[i], shuffledArray[j]] = [shuffledArray[j], shuffledArray[i]];
+  }
+  return shuffledArray;
+};
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/core/utils/array.shuffle.ts:8]: remind me some other time

[the remark this answers is not in the document]
Comment (hero101) [reply src/core/utils/array.shuffle.ts:8]: yes

**Review by bobbykolev — DISMISSED** [review]
The code looks good and the implementation works. 
Note that with the current server/API implementation you can't see users without tags (even using the search). This could be misleading but not related to this PR.

**Review by bobbykolev — APPROVED** [review]

**Pull request merged**